### PR TITLE
Remove CRTP from psibase::Service

### PR DESCRIPTION
**Pull request by swatanabe**
- Update to C++23
- Ubuntu 20.04 is no longer supported

### CODE REVIEW

**Comment by swatanabe** [comment]
Requires GCC 14. Ubuntu 24.04 has GCC 13.

File: libraries/psibase/common/include/psibase/Service.hpp
Comment (James-Mart) [inline libraries/psibase/common/include/psibase/Service.hpp:1]: Without the crtp, I'm not sure there's much benefit to the inheritance. Why not just move `emit` and `events` to a separate events library?

Comment (swatanabe) [reply libraries/psibase/common/include/psibase/Service.hpp:1]: I didn't change the syntax of calling `emit()`. A free function would need the service to be passed explicitly.

Comment (swatanabe) [reply libraries/psibase/common/include/psibase/Service.hpp:1]: Deducing `this` can replace most uses of CRTP.

**Review by James-Mart — APPROVED** [review]

**Pull request merged**